org-code-javabuilder/lib/src/test/java/org/code/javabuilder/UserProjectFileParserTest.java

```diff
@@ -39,4 +39,26 @@ public void throwsExceptionOnInvalidJson() throws UserFacingException, UserIniti
     assertThrows(
         UserFacingException.class, () -> this.userProjectFileParser.parseFileJson(invalidJson));
   }
+
+  @Test
+  public void canParseMultipleFiles() throws UserFacingException, UserInitiatedException {
+    String validJson =
+        "{\n"
+            + "  \"source\": {\n"
+            + "    \"ScannerSample.java\": {\n"
+            + "      \"text\": \"import java.util.Scanner;\\n\\npublic class ScannerSample {\\n\\tpublic static void main(String[] args) {\\n\\t\\tHelperClass.printWelcomeMessage();\\n\\t\\tScanner console = new Scanner(System.in);\\n\\t\\tSystem.out.println(\\\"Enter your name: \\\");\\n\\t\\tString name = console.next();\\n\\t\\tSystem.out.println(\\\"Hello \\\" + name);\\n\\t}\\n}\\n\",\n"
+            + "      \"visible\": true\n"
+            + "    },\n"
+            + "    \"HelperClass.java\": {\n"
+            + "      \"text\": \"\\npublic class HelperClass {\\n\\tpublic static void printWelcomeMessage() {\\n\\t\\tSystem.out.println(\\\"Hello!\\\");\\n\\t}\\n}\\n\",\n"
+            + "      \"visible\": true\n"
+            + "    }\n"
+            + "  },\n"
+            + "  \"animations\": {}\n"
+            + "}";
```

Comment:
Suggestion: Since this test is testing the file names and size, omit unneeded sections. i.e. `text` could be something short like `"my code"`.